src/category_theory/monoidal/braided.lean

```diff
@@ -0,0 +1,116 @@
+extend this. The rationale is that we are not carrying any additional data,
+just requiring a property.
+
+## Future work
+
+* Construct the Drinfeld center of a monoidal category as a braided monoidal category.
+* Say something about pseudo-natural transformations.
+
+-/
+
+open category_theory
+
+universes v v₁ v₂ v₃ u u₁ u₂ u₃
+
+namespace category_theory
+
+/--
+A braided monoidal category is a monoidal category equipped with a braiding isomorphism
+`β_ X Y : X ⊗ Y ≅ Y ⊗ X`
+which is natural in both arguments,
+and also satisfies the two hexagon identities.
+-/
+class braided_category (C : Type u) [category.{v} C] [monoidal_category.{v} C] :=
+-- braiding natural iso:
+(braiding             : Π X Y : C, X ⊗ Y ≅ Y ⊗ X)
+(braiding_naturality' : ∀ {X X' Y Y' : C} (f : X ⟶ Y) (g : X' ⟶ Y'),
+  (f ⊗ g) ≫ (braiding Y Y').hom = (braiding X X').hom ≫ (g ⊗ f) . obviously)
+-- hexagon identities:
+(hexagon_forward'     : Π X Y Z : C,
+    (α_ X Y Z).hom ≫ (braiding X (Y ⊗ Z)).hom ≫ (α_ Y Z X).hom
+  = ((braiding X Y).hom ⊗ (𝟙 Z)) ≫ (α_ Y X Z).hom ≫ ((𝟙 Y) ⊗ (braiding X Z).hom)
+  . obviously)
+(hexagon_reverse'     : Π X Y Z : C,
+    (α_ X Y Z).inv ≫ (braiding (X ⊗ Y) Z).hom ≫ (α_ Z X Y).inv
+  = ((𝟙 X) ⊗ (braiding Y Z).hom) ≫ (α_ X Z Y).inv ≫ ((braiding X Z).hom ⊗ (𝟙 Y))
+  . obviously)
+
+restate_axiom braided_category.braiding_naturality'
+attribute [simp] braided_category.braiding_naturality
+restate_axiom braided_category.hexagon_forward'
+attribute [simp] braided_category.hexagon_forward
+restate_axiom braided_category.hexagon_reverse'
+attribute [simp] braided_category.hexagon_reverse
+
+open braided_category
+
+notation `β_` := braiding
+
+section prio
+set_option default_priority 100 -- see Note [default priority]
+
+/--
+A symmetric monoidal category is a braided monoidal category for which the braiding is symmetric.
+-/
+class symmetric_category (C : Type u) [category.{v} C] [monoidal_category.{v} C]
+   extends braided_category.{v} C :=
+-- braiding symmetric:
+(symmetry' : ∀ X Y : C, (β_ X Y).hom ≫ (β_ Y X).hom = 𝟙 (X ⊗ Y) . obviously)
+
+end prio
+
+restate_axiom symmetric_category.symmetry'
+attribute [simp] symmetric_category.symmetry
+
+variables (C : Type u₁) [category.{v₁} C] [monoidal_category C] [braided_category C]
+variables (D : Type u₂) [category.{v₂} D] [monoidal_category D] [braided_category D]
+variables (E : Type u₃) [category.{v₃} E] [monoidal_category E] [braided_category E]
+
+/--
+A braided functor between braided monoidal categories is a monoidal functor
+which preserves the braiding.
+-/
+structure braided_functor extends monoidal_functor C D :=
+-- I have no reason to think this formulation is the best, in terms of `simp` normal forms.
+-- Suggestions for moving the `μ`s around, or swapping left- and right-hand sides, very welcome.
+(braided' : ∀ X Y : C, map (β_ X Y).hom = inv (μ X Y) ≫ (β_ (obj X) (obj Y)).hom ≫ μ Y X . obviously)
```

Comment:
Fwiw, my instinct is that this is the right formulation, since it expands out the `map`, which is arguably the least-simplified bit, and the fact that composition below just worked seems to suggest this too

Reply:
Great, okay, replaced with a comment explaining the choice of simp normal form.